carina-utils/src/main/java/com/qaprosoft/carina/core/foundation/utils/naming/TestNamingUtil.java

```diff
@@ -167,7 +168,14 @@ public static String getCanonicalTestName(ITestResult result) {
             }
         }
 
-        return StringEscapeUtils.escapeHtml4(appendTestMethodName(testName, result.getMethod()));
+        testName = StringEscapeUtils.escapeHtml4(appendTestMethodName(testName, result.getMethod()));
+
+        String customCapability = Configuration.get(Configuration.Parameter.CUSTOM_CAPABILITIES);
+        if (!customCapability.isEmpty() && customCapability.toLowerCase().contains("browserstack")) {
```

Comment:
R.CONFIG is shared across all threads so this change could rename a lot of tetsMethods for browserstack if they started in multi-threading mode

Reply:
Got it, will update this one.